tools/clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -7759,6 +7763,14 @@ def warn_hlsl_unary_negate_unsigned : Warning<
 def warn_hlsl_impcast_glc_mismatch : Warning<
   "implicit conversion from %0 to %1 %select{loses|adds}2 globallycoherent annotation">,
   InGroup<Conversion>, DefaultWarn;
+def warn_hlsl_impcast_rdc_mismatch : Warning<
```

Comment:
We should also combine this with `warn_hlsl_impcast_glc_mismatch`, and remove the `DefaultWarn` marker.

Bonus points if we can also fold `warn_hlsl_impcast_rdc_glc_mismatch` into the same warning.

Reply:
Fused all coherence mismatch diagnostics into one